google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardLibraryPanel.java

```diff
@@ -0,0 +1,121 @@
+ */
+public class AppEngineStandardLibraryPanel {
+
+  private JPanel mainPanel;
+  private JPanel libraryPanel;
+
+  private boolean enabled = true;
+
+  public AppEngineStandardLibraryPanel(boolean enabled) {
+    this.enabled = enabled;
+  }
+
+  public Set<AppEngineStandardMavenLibrary> getSelectedLibraries() {
+    return Sets.newHashSet(Collections2.filter(
+        Collections2.transform(Arrays.asList(libraryPanel.getComponents()),
+            new Function<Component, AppEngineStandardMavenLibrary>() {
+              @Nullable
+              @Override
+              public AppEngineStandardMavenLibrary apply(Component libraryCheckbox) {
+                return libraryCheckbox instanceof JCheckBox
+                    && ((JCheckBox) libraryCheckbox).isSelected()
+                    ? AppEngineStandardMavenLibrary
+                    .getLibraryByDisplayName(((JCheckBox) libraryCheckbox).getText())
+                    : null;
+              }
+            }),
+        Predicates.notNull()
+    ));
+  }
+
+  public void setSelectLibraries(Set<AppEngineStandardMavenLibrary> libraries) {
+    Collection<String> availableLibraryNames = Collections2.transform(libraries,
+        new Function<AppEngineStandardMavenLibrary, String>() {
+          @Nullable
+          @Override
+          public String apply(AppEngineStandardMavenLibrary library) {
+            return library.getDisplayName();
+          }
+        });
+
+    for (Component libraryCheckbox : libraryPanel.getComponents()) {
+      ((JCheckBox) libraryCheckbox)
+          .setSelected(availableLibraryNames.contains(((JCheckBox) libraryCheckbox).getText()));
+    }
+  }
+
+  public void toggleLibrary(AppEngineStandardMavenLibrary library, boolean select) {
+    for (Component libraryCheckbox : libraryPanel.getComponents()) {
+      if (libraryCheckbox instanceof JCheckBox
+          && ((JCheckBox) libraryCheckbox).getText().equals(library.getDisplayName())) {
+        ((JCheckBox) libraryCheckbox).setSelected(select);
+      }
+    }
+  }
+
+  @NotNull
+  public JPanel getComponent() {
+    return libraryPanel;
+  }
+
+  public boolean isEnabled() {
+    return enabled;
+  }
+
+  @SuppressWarnings("checkstyle:abbreviationaswordinname")
+  private void createUIComponents() {
+    libraryPanel = new JPanel(new GridLayout(AppEngineStandardMavenLibrary.values().length, 1));
+    if (enabled) {
+      libraryPanel.setBorder(
+          BorderFactory.createTitledBorder(BorderFactory.createEtchedBorder(), "Libraries"));
+
+      for (AppEngineStandardMavenLibrary library : AppEngineStandardMavenLibrary.values()) {
+        final JCheckBox libraryCheckbox = new JCheckBox(library.getDisplayName());
+        libraryPanel.add(libraryCheckbox);
+
+        // The Servlet API is provided by the AE standard runtime. So we are enabling it by default.
+        if (library == AppEngineStandardMavenLibrary.SERVLET_API) {
+          libraryCheckbox.setSelected(true);
```

Comment:
I was going to not allow disabling, but this made the experience for project imports confusing:

If the user has an existing project, then adds our facet (makes in an AE standard app), then the lib configuration for the servlet would have to be enabled in this case to allow them to add it. It felt cleaner just to leave it editable all the time given this.

That being said, its selected by default in the new project wizard.